src/test/kotlin/no/nav/tilleggsstonader/sak/vedtak/barnetilsyn/beregning/TilsynBarnVedtaksperiodeValidingerUtilsTest.kt

```diff
@@ -0,0 +1,781 @@
+import java.time.LocalDate
+import java.time.YearMonth
+
+class TilsynBarnVedtaksperiodeValidingerUtilsTest {
+    val vilkårperiodeService = mockk<VilkårperiodeService>()
+    val grunnlagsdataService = mockk<GrunnlagsdataService>()
+    val tilsynBarnVedtaksperiodeValidingerService =
+        TilsynBarnVedtaksperiodeValidingerService(
+            vilkårperiodeService = vilkårperiodeService,
+            grunnlagsdataService = grunnlagsdataService,
+        )
+
+    val behandlingId = BehandlingId.random()
+
+    val målgrupper =
+        listOf(
+            målgruppe(
+                faktaOgVurdering = faktaOgVurderingMålgruppe(type = MålgruppeType.AAP),
+                fom = LocalDate.of(2025, 1, 1),
+                tom = LocalDate.of(2025, 2, 28),
+            ),
+        )
+    val aktiviteter =
+        listOf(
+            aktivitet(
+                faktaOgVurdering = faktaOgVurderingAktivitetTilsynBarn(type = AktivitetType.TILTAK),
+                fom = LocalDate.of(2025, 1, 1),
+                tom = LocalDate.of(2025, 2, 28),
+            ),
+        )
+
+    @BeforeEach
+    fun setup() {
+        every { vilkårperiodeService.hentVilkårperioder(any()) } returns
+            Vilkårperioder(
+                målgrupper = målgrupper,
+                aktiviteter = aktiviteter,
+            )
+        every { grunnlagsdataService.hentGrunnlagsdata(any()) } returns lagGrunnlagsdata()
+    }
+
+    @Test
+    fun `skal ikke kaste feil for gyldig vedtaksperiode`() {
+        val vedtaksperiode = lagVedtaksperiode()
+
+        assertThatCode {
+            tilsynBarnVedtaksperiodeValidingerService.validerVedtaksperioder(
+                listOf(
+                    vedtaksperiode,
+                ),
+                behandlingId,
+            )
+        }.doesNotThrowAnyException()
+    }
+
+    @Test
+    fun `skal kaste feil om kombinasjon av målgruppe og aktivitet er ugyldig`() {
+        val vedtaksperiode =
+            lagVedtaksperiode(målgruppe = MålgruppeType.OVERGANGSSTØNAD, aktivitet = AktivitetType.TILTAK)
+
+        assertThatThrownBy {
+            tilsynBarnVedtaksperiodeValidingerService.validerVedtaksperioder(
+                listOf(
+                    vedtaksperiode,
+                ),
+                behandlingId,
+            )
+        }.hasMessageContaining("Kombinasjonen av ${vedtaksperiode.målgruppe} og ${vedtaksperiode.aktivitet} er ikke gyldig")
+    }
+
+    @Test
+    fun `skal kaste feil om ingen periode for målgruppe matcher`() {
+        val vedtaksperiode = lagVedtaksperiode(målgruppe = MålgruppeType.NEDSATT_ARBEIDSEVNE)
+
+        assertThatThrownBy {
+            tilsynBarnVedtaksperiodeValidingerService.validerVedtaksperioder(
+                listOf(
+                    vedtaksperiode,
+                ),
+                behandlingId,
+            )
+        }.hasMessageContaining("Finner ingen perioder hvor vilkår for ${vedtaksperiode.målgruppe} er oppfylt")
```

Comment:
Her vil vel alltid ${vedtaksperiode.målgruppe) være "NEDSATT_ARBEIDSEVNE" ? kunne feilmeldingen bli skrevet ut i klartekst her, for tydelighetens skyld?

Reply:
dvs:
`Finner ingen perioder hvor vilkår for NEDSATT_ARBEIDSEVNE er oppfylt`

Reply:
Denne kommentaren gjelder for alle stedene i denne filen hvor tilsvarende programmatisk resolving av verdier gjøres

Reply:
Jepsi! 
6f071bedcd849912858b946cd10e09c2ef55f074